pkg/cloud/aws/services/userdata/controlplane.go

```diff
@@ -21,11 +21,22 @@ import "github.com/pkg/errors"
 const (
 	controlPlaneBashScript = `{{.Header}}
 
-mkdir -p /etc/kubernetes/pki
+set -eox
+
+mkdir -p /etc/kubernetes/pki/etcd
 
 echo '{{.CACert}}' > /etc/kubernetes/pki/ca.crt
 echo '{{.CAKey}}' > /etc/kubernetes/pki/ca.key
```

Comment:
keys need to be written with `600` permissions to match what kubeadm does

Reply:
it would also be good to have a -n on the echo so as not to have an additional new line in the certificate and key files